applications/CompressiblePotentialFlowApplication/python_scripts/initialize_geometry.py

```diff
@@ -0,0 +1,252 @@
+            In order to perform the refinement, it is needed to calculate the distance gradient, the initial nodal_h and the level_set metric.
+        '''
+        ini_time=time.time()
+        local_gradient = KratosMultiphysics.ComputeNodalGradientProcess2D(self.main_model_part, KratosMultiphysics.DISTANCE,
+                                            KratosMultiphysics.DISTANCE_GRADIENT, KratosMultiphysics.NODAL_AREA)
+        local_gradient.Execute()
+
+        find_nodal_h = KratosMultiphysics.FindNodalHNonHistoricalProcess(self.main_model_part)
+        find_nodal_h.Execute()
+
+        metric_process = MeshingApplication.ComputeLevelSetSolMetricProcess2D(self.main_model_part,  KratosMultiphysics.DISTANCE_GRADIENT, self.metric_parameters)
+        metric_process.Execute()
+
+        self.PrintOutput('metric_output'+str(self.step))
+
+        mmg_parameters = KratosMultiphysics.Parameters("""
+        {
+            "discretization_type"                  : "STANDARD",
+            "save_external_files"              : false,
+            "initialize_entities"              : false,
+            "echo_level"                       : 0
+        }
+        """)
+
+        mmg_process = MeshingApplication.MmgProcess2D(self.main_model_part, mmg_parameters)
+        mmg_process.Execute()
+
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','Remesh time: ',time.time()-ini_time)
+        self.PrintOutput('remeshed_output'+str(self.step))
+
+    def UpdateParameters(self):
+        ''' This process updates remeshing parameters in case more than one iteration is performed'''
+        previous_size=self.metric_parameters["minimal_size"].GetDouble()
+        self.metric_parameters["minimal_size"].SetDouble(previous_size*0.5)
+
+    def ModifyFinalDistance(self):
+        ''' This function modifies the distance field to avoid ill defined cuts.
+        '''
+        ini_time = time.time()
+        find_nodal_h = KratosMultiphysics.FindNodalHProcess(self.main_model_part)
+        find_nodal_h.Execute()
+        KratosMultiphysics.FluidDynamicsApplication.DistanceModificationProcess(self.main_model_part,self.distance_modification_parameters).Execute()
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','Modify distance time: ',time.time()-ini_time)
+
+    def CopyAndDeleteDefaultDistance(self):
+        ''' This function copies the distance field to an auxiliary distance variable and sets
+        to zero the default one.
+        '''
+        KratosMultiphysics.VariableUtils().CopyScalarVar(KratosMultiphysics.DISTANCE,CompressiblePotentialFlow.GEOMETRY_DISTANCE, self.main_model_part.Nodes)
+        KratosMultiphysics.VariableUtils().SetHistoricalVariableToZero(KratosMultiphysics.DISTANCE, self.main_model_part.Nodes)
+
+    def PrintOutput(self,filename):
+        if self.print_output_flag:
+            from gid_output_process import GiDOutputProcess
+            gid_output = GiDOutputProcess(self.main_model_part,
```

Comment:
Which is the intention of having the GiDOutputProcess encapsulated in here?. As far as I understand, this process is intended to refine a coarse background mesh according to the level-set function.

Reply:
This is used to debug/test cases. It is useful to check the metric values at each step of the remeshing process to check if you would need more or less iterations. Or to check the effect of a good/bad background mesh. 

I can remove this to just use it in case I need it but maybe others can benefit from it when developing other formulations or 3D cases. What would you do?

Reply:
If it is for debugging it shouldn't be in the master. If you want to keep it for further developments (e.g. 3D) I'd commented so it is not executed by default each time the process is called.

Assuming that each one of the intermediate meshes is saved in a different model part (I don't know if this is the case), you can print them from outside the process by setting a GiD output process for each one of them in the json settings `output_processes` section.

Reply:
I understand!
What you are saying about the different model parts could be interesting! I will give it a try.  If i save each intermediate mesh, does it slow in some way the whole simulation?

Reply:
I will not slow it but the memory consumption will significantly increase. But taking into account that you need this for debugging 2D cases (so far) it wouldn't be that much. Of course this is a temporary alternative solution.